tests/utils/test_archive_scientific_readers.py

```
@@ -473,3 +473,570 @@ def test_corrupted_hdf5_file(self, tmp_path: Path) -> None:
+    def test_read_zip_file_via_fileobj_default_label(self, sample_zip_file: Path) -> None:
+        """read_zip_file() uses '<fileobj>' label when no file_path given."""
+        with sample_zip_file.open("rb") as f:
+            result = read_zip_file(fileobj=f)
+        assert "<fileobj>" in result
+
+    def test_read_zip_file_via_fileobj_corrupted_raises(self, tmp_path: Path) -> None:
+        """read_zip_file() with a bad fileobj raises FileReadError."""
+        bad = tmp_path / "bad.zip"
+        bad.write_bytes(b"not a zip")
+        with bad.open("rb") as f:
+            with pytest.raises(FileReadError):
```

Ruff (0.15.13)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)